trl/trainer/sft_trainer.py

```diff
@@ -396,12 +396,15 @@ def _prepare_dataset(
         remove_unused_columns=True,
         append_concat_token=True,
         add_special_tokens=True,
+        skip_prepare_dataset=False,
     ):
         if dataset is None:
             raise ValueError("The dataset should not be None")
 
         # check if torch dataset / dataloader and do nothing
-        if isinstance(dataset, (torch.utils.data.IterableDataset, torch.utils.data.Dataset, ConstantLengthDataset)):
+        if skip_prepare_dataset or isinstance(
```

Comment:
Let's add an integration test for this case under the existing `SFTTrainer` tests, along with an example training a tiny random llava model